File: dotnet/Services/TodoService.cs
using Todo.Core.Abstract.Repositories;
using Todo.Core.Abstract.Services;
using Todo.Core.Exceptions;
using Todo.Core.Models;

namespace Todo.Services;

public class TodoService(ITodoRepository todoRepository) : ITodoService
{
    public Task<TodoEntity> Create(TodoEntity entity)
        => todoRepository.Create(entity);

    public Task<ICollection<TodoEntity>> GetAll()
        => todoRepository.GetAll();


    public async Task SetDone(int id)
    {
        var isDeleted = await todoRepository.Delete(id);

        if (!isDeleted)
            throw new TodoNotFoundException(id);
    }
}

File: nodejs/src/services/todo-service.ts
import { ITodoRepository } from "../core/abstract/repositories/todo-repository";
import { ITodoService } from "../core/abstract/services/todo-service";
import { TodoNotFoundError } from "../core/errors/todo-not-found.error";
import { Todo } from "../core/todo";

export class TodoService implements ITodoService {
  constructor(private readonly todoRepository: ITodoRepository) {}

  create(todo: Todo): Promise<Todo> {
    return this.todoRepository.create(todo);
  }

  getAll(): Promise<Todo[]> {
    return this.todoRepository.getAll();
  }

  setIsDone(id: number): Promise<void> {
    throw new Error("");
  }

  async delete(id: number): Promise<void> {
    const isDeleted = await this.todoRepository.delete(id);

    if (!isDeleted) {
      throw new TodoNotFoundError(id);
    }
  }
}
